### PR TITLE
Add detekt

### LINKED ISSUE #26: Add Detekt to Travis CI
What about introducing the lib in travis?

https://github.com/arturbosch/detekt

### PR DESCRIPTION
**Pull request by JMPergar**
Fixes #26

### CODE REVIEW

File: detekt.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    active: true
+    autoCorrect: true
+  TrailingSpaces:
+    active: true
+    autoCorrect: true
+  UnusedImports:
+    active: true
+    autoCorrect: true
+  OptionalSemicolon:
+    active: true
```
Comment (ffgiraldez) [inline detekt.yml]: I Like fail if some use ; it's that the config or the opposite?

File: katz/src/main/kotlin/katz/typeclasses/MonadError.kt [hunk 2]
```diff
@@ -46,7 +46,7 @@ class MonadErrorContinuation<F, A>(val ME : MonadError<F, Throwable>) : Serializ
  */
 fun <F, B> MonadError<F, Throwable>.binding(c: suspend MonadErrorContinuation<F, *>.() -> HK<F, B>): HK<F, B> {
     val continuation = MonadErrorContinuation<F, B>(this)
-    val f: suspend MonadErrorContinuation<F, *>.() -> HK<F, B> = { c() }
+    val f: suspend MonadErrorContinuation < F, * > .() -> HK < F, B > = { c() }
```
Comment (ffgiraldez) [inline katz/src/main/kotlin/katz/typeclasses/MonadError.kt]: I prefer the previous format, could ve this posible?

Comment (JMPergar) [inline katz/src/main/kotlin/katz/typeclasses/MonadError.kt]: I'm working on review all rules, I prefer the previous format too.

File: detekt.yml (hunk 1 above)
Comment (JMPergar) [inline detekt.yml]: I'll check it. I prefer it too.

**Pull request merged**